languagetool-core/src/test/java/org/languagetool/language/NGramLangIdTest.java

```diff
@@ -0,0 +1,171 @@
+package org.languagetool.language;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.*;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.zip.ZipFile;
+
+import static org.junit.Assert.*;
+import static org.junit.Assert.assertTrue;
+
+public class NGramLangIdTest {
+  public NGramLangIdTest() throws IOException {
+  }
+
+  private class ProcessedSentence {
+    public final String sentence;
+    public final List<Integer> encoding;
+    public final List<Double> pred;
+
+    public ProcessedSentence(String part1, String part2, String part3) {
+     sentence = part1;
+     encoding = Arrays.stream(part2.split(" ")).map(Integer::parseInt).collect(Collectors.toList());
+     pred = Arrays.stream(part3.split(" ")).map(Double::parseDouble).collect(Collectors.toList());
+    }
+  }
+
+  private final File TESTER_MODEL = new File("C:\\Users\\Robert\\Desktop\\ngram-language-identification\\testing_model\\model\\model.zip");
```

Comment:
How large is this file? Should it be in git? If not, the test will need to be `@Ignore`.

Reply:
Yes, the file contains a trimmed model (~1.5MB) that is extracted from the main one - I'll add it to git once I got the final file. I noticed that the current model with a vocabulary of 30k is too big/slow, so I'm reducing it.